report/firefox_code_coverage/codecoverage.py

```diff
@@ -69,27 +70,28 @@ def _save_tasks(response):
 
 def download_binary(url, path, retries=5):
     """Download a binary file from an url"""
-    for i in range(1, retries + 1):
+    @tenacity.retry(stop=tenacity.stop_after_attempt(retries),
+                    wait=tenacity.wait_incrementing(start = 7, increment = 7),
+                    reraise=True)
+    def retry_function():
         try:
             artifact = requests.get(url, stream=True)
             artifact.raise_for_status()
 
             with open(path, "wb") as f:
                 for chunk in artifact.iter_content(chunk_size=8192):
                     f.write(chunk)
-            break
-        except:  # noqa: E722
+        except:
             try:
                 os.remove(path)
             except OSError:
                 pass
-
-            if i == retries:
-                raise Exception(
-                    "Download failed after {} retries - {}".format(retries, url)
+            
+            raise Exception(
+                    "Download failed after {} retries - {}".format(retry_function.retry.statistics["attempt_number"], url)
```

Comment:
We can remove the number of retries from the exception message to simplify this code.

Reply:
Even more simply, just do `raise` to re-raise the previous exception.

Reply:
i have made this change